### PR TITLE
add megrez strategy and algorithm

### PR DESCRIPTION
**Pull request by tianpf**
add megrez  strategy  and algorithm

### CODE REVIEW

File: algorithm/matrix/matrix.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
+package matrix
+
+// Matrix is a one-dimensional matrix.
+type Matrix []float64
+
+// LineMatrix is a two-dimensional matrix.
+type LineMatrix [][]float64
+
+// PolygonMatrix is a there-dimensional matrix.
```
Comment (zhyt1985) [inline algorithm/matrix/matrix.go]: A typo, "there-dimensional" should be "three-dimensional"

File: bound.go [hunk 2]
```diff
@@ -83,7 +83,57 @@ func (b Bound) Bound() Bound {
...
+func (b Bound) Equal(g Geometry) bool {
+	if g.GeoJSONType() != b.GeoJSONType() {
+		return false
+	}
+	return b.EqualBound(g.(Bound))
+}
+
+// Area returns the area of a polygonal geometry. The area of a bound is 0.
+func (b Bound) Area() (float64, error) {
+	return 0.0, nil
```
Comment (zhyt1985) [inline bound.go]: The area of a bound is not always 0

File: multi_point.go [hunk 3]
```diff
@@ -1,7 +1,9 @@
 package geoos
 
+import "github.com/spatial-go/geoos/algorithm/matrix"
+
 // A MultiPoint represents a set of points in the 2D Eucledian or Cartesian plane.
-type MultiPoint []Point
+type MultiPoint matrix.LineMatrix
```
Comment (zhyt1985) [inline multi_point.go]: I think `MultiPoint` should be defined as []Point

Comment (zhyt1985) [reply multi_point.go]: `MultiLineString` is defined as []LineString, `MultiPolygon` is defined as []Polygon

File: algorithm/matrix/matrix.go (hunk 1 above)
Comment (tianpf) [reply algorithm/matrix/matrix.go]: take and done

File: bound.go (hunk 2 above)
Comment (tianpf) [reply bound.go]: take and done

File: multi_point.go (hunk 3 above)
Comment (tianpf) [reply multi_point.go]: take and done

**Review by zhyt1985 — APPROVED** [review]

File: .vscode/settings.json [hunk 4]
```diff
@@ -6,5 +6,6 @@
         "stdio.h": "c",
```
Comment (xiaogu-space) [inline .vscode/settings.json]: .vscode folder don't upload

File: algorithm/area.go [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+			area += Area(ring)
+		} else {
+			area -= Area(ring)
+		}
+	}
+	return area
+}
+
+// Area returns the area of a Ring geometry.
+func Area(ring matrix.LineMatrix) float64 {
```
Comment (xiaogu-space) [inline algorithm/area.go:32]: Add algorithm description url

Comment (tianpf) [reply algorithm/area.go:32]: Unified to add the algorithm description

**Review by xiaogu-space — APPROVED** [review]

**Pull request merged**